docs/manual/comparison_with_sql.rst

```diff
@@ -0,0 +1,662 @@
+                  Grapes,  10/7/2016, Bob,    12
+                  Grapes,  10/7/2016,  Tony,   15
+
+.. code-block:: SQL
+
+    SELECT fruit,
+           name,
+           SUM(number) as sum_num
+    FROM fruits_data
+    GROUP BY fruit, name;
+
+
+.. code-block:: python
+
+    # datatable
+    fruits_data[:,
+                {"sum_num": dt.sum(f.Number)},
+                by('Fruit', 'Name')]
+
+.. csv-table:: Aggregations on Multiple COlumns
+    :header: Fruit, Name, sum_num
+    :widths: 10,10, 10
+
+    Apples,	Bob,	16
+	Apples,	Mike,	9
+	Apples,	Steve,	10
+	Grapes,	Bob,	35
+	Grapes,	Tom,	87
+	Grapes,	Tony,	15
+	Oranges,	Bob,	67
+	Oranges,	Mike,	57
+	Oranges,	Tom,	15
+	Oranges,	Tony,	1
+
+- We can replicate SQL's ``WHERE`` clause in a ``GROUP BY``
+
+.. code-block:: SQL
+
+    SELECT species,
+           AVG(sepal_length) as avg_sepal_length,
+           SUM(number) as sum_num
+    FROM iris
+    WHERE sepal_width > 3
+    GROUP BY species;
+
+
+.. code-block:: python
+
+    # datatable
+    iris[f.sepal_width >=3, :][:,
+                              {"avg_sepal_length": dt.mean(f.sepal_length)},
+                              by('species')]
+
+.. csv-table:: Filtration in a Group By
+    :header: species, avg_sepal_length
+    :widths: 10,10
+
+    setosa,	5.02917
+    versicolor,	6.21875
+	virginica,	6.76897
+
+- We can also replicate SQL's ``HAVING`` clause in a ``GROUP BY``
+
+.. code-block:: SQL
+
+    SELECT fruit,
+           name,
+           SUM(number) as sum_num
+    FROM fruits_data
+    GROUP BY fruit, name
+    HAVING sum_num > 50;
+
+
+.. code-block:: python
+
+    # datatable
+    fruits_data[:,
+               {'sum_num': dt.sum(f.Number)},
+               by('Fruit','Name')][f.sum_num > 50, :]
+
+.. csv-table:: Filtration after a Group By
+    :header: Fruit, Name, sum_num
+    :widths: 10,10, 10
+
+    Grapes,	Tom,	87
+	Oranges,	Bob,	67
+	Oranges,	Mike,	57
+
+
+- Grouping on a condition
+
+.. code-block:: SQL
+
+    SELECT sepal_width >=3 as width_larger_than_3,
+           AVG(sepal_length) as avg_sepal_length
+    FROM iris
+    GROUP BY sepal_width>=3;
+
+
+.. code-block:: python
+
+    # datatable
+    iris[:,
+         {"avg_sepal_length": dt.mean(f.sepal_length)},
+         by(f.sepal_width >= 3)]
+
+.. csv-table:: Grouping on a Condition
+    :header: CO, avg_sepal_length
+    :widths: 10,10
+
+    	0,	5.95263
+    	1,	5.77634
+
+At the moment, names cannot be assigned in the ``by`` section.
+
+LEFT OUTER JOIN
+----------------
+
+We will compare the left outer join, as that is the only join currently implemented in ``datatable``. Another aspect is that the dataframe being joined must be keyed, the column or columns to be keyed must not have duplicates, and the joining column has to have the same name in both dataframes. You can read more about the :func:`join()` API and have a look at the `Tutorial on the join operator <https://datatable.readthedocs.io/en/latest/start/quick-start.html#join>`_
```

Comment:
I guess you can just say `frame` instead of a `dataframe` to be consistent with the rest of the docs we have now. It will also be more intuitive for the user to establish a relation `frame` <-> `dt.Frame()`, and not do something like `dt.dataframe()` that won' work...